hack/vagrant/script/install_kubernetes.sh

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBERNETES_VERSION="v1.10.2"
+CRI_VERSION="v1alpha2"
+CNI_VERSION="v0.6.0"
+MASTER_CIDR="10.244.0.0/16"
+
+apt-get install -y ebtables socat
+
+# Configuring pouch's service to enable cri for kubelet
+mkdir -p /etc/systemd/system/pouch.service.d
+cat > /etc/systemd/system/pouch.service.d/pouch.conf <<EOF
+[Service]
+ExecStart=
+ExecStart=/usr/bin/pouchd --enable-cri=true --cri-version=$CRI_VERSION
+EOF
+systemctl daemon-reload
+systemctl restart pouch
+
+# Installing kubernetes's components
+wget -O- --read-timeout=20 --progress=bar:force:noscroll https://raw.githubusercontent.com/manlge/resources/master/kubernetes/$KUBERNETES_VERSION/kubernetes-linux-amd64.tar.xz|tar -C /usr/bin -xJ
+chmod +x /usr/bin/kubeadm /usr/bin/kubelet /usr/bin/kubectl
+
+KUBELET_URL="https://raw.githubusercontent.com/kubernetes/kubernetes/$KUBERNETES_VERSION/build/debs"
+mkdir -p /etc/systemd/system/kubelet.service.d
+curl -sS "$KUBELET_URL/kubelet.service" -o /etc/systemd/system/kubelet.service
+curl -sS "$KUBELET_URL/10-kubeadm.conf" -o /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+sed -i '2 i\Environment="KUBELET_EXTRA_ARGS=--container-runtime=remote --container-runtime-endpoint=unix:///var/run/pouchcri.sock --image-service-endpoint=unix:///var/run/pouchcri.sock"' /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+systemctl enable kubelet
+systemctl daemon-reload
+systemctl start kubelet
+
+# Installing CNI
+mkdir -p /opt/cni/bin
+wget -O- --read-timeout=20 --progress=bar:force:noscroll "https://raw.githubusercontent.com/manlge/resources/master/cni/$CNI_VERSION/cni-plugins-amd64-$CNI_VERSION.tgz" | tar -C /opt/cni/bin -xz
+
+# Create a single-host Kubernetes cluster by kubeadm init
+KUBEADM_CONFIG=/tmp/kubeadm.yaml
+cat <<EOF > $KUBEADM_CONFIG
+apiVersion: kubeadm.k8s.io/v1alpha1
+kind: MasterConfiguration
+kubernetesVersion: $KUBERNETES_VERSION
+api:
+  bindPort: 6443
+certificatesDir: /etc/kubernetes/pki
+clusterName: pouch
+imageRepository: registry.cn-hangzhou.aliyuncs.com/google_containers
+networking:
+  dnsDomain: cluster.local
+  serviceSubnet: 10.96.0.0/12
+  podSubnet: $MASTER_CIDR
+nodeRegistration:
+  criSocket: /var/run/pouchcri.sock
+EOF
+kubeadm init --config $KUBEADM_CONFIG --ignore-preflight-errors=all
+
+# Config default kubeconfig for kubectl
+mkdir -p $HOME/.kube
+cat /etc/kubernetes/admin.conf > $HOME/.kube/config
+chown $(id -u):$(id -g) $HOME/.kube/config
+
+until kubectl get nodes &> /dev/null; do echo "Waiting kubernetes for a second..."; sleep 1; done
```

Comment:
could we consider to put this file into hack folder, because it might be used out of the vagrant?